service/testutil.go

```diff
@@ -0,0 +1,232 @@
+package service
```

Comment:
I think it best to take all of the testing stuff and put it into a `test` or `testing` package by itself, not in the `service` package.